src/VisualStudio/Core/Def/Options/VisualStudioSettingsOptionPersister.cs

```diff
@@ -45,20 +37,14 @@ private class SVsSettingsPersistenceManager { };
         /// <remarks>
         /// We make sure this code is from the UI by asking for all <see cref="IOptionPersister"/> in <see cref="RoslynPackage.InitializeAsync"/>
         /// </remarks>
-        public VisualStudioSettingsOptionPersister(Action<OptionKey2, object?> refreshOption, ImmutableDictionary<string, Lazy<IVisualStudioStorageReadFallback, OptionNameMetadata>> readFallbacks, ISettingsManager? settingsManager)
+        public VisualStudioSettingsOptionPersister(Action<OptionKey2, object?> refreshOption, ImmutableDictionary<string, Lazy<IVisualStudioStorageReadFallback, OptionNameMetadata>> readFallbacks, ISettingsManager settingsManager)
         {
             _settingManager = settingsManager;
             _refreshOption = refreshOption;
             _readFallbacks = readFallbacks;
 
-            // While the settings persistence service should be available in all SKUs it is possible an ISO shell author has undefined the
-            // contributing package. In that case persistence of settings won't work (we don't bother with a backup solution for persistence
-            // as the scenario seems exceedingly unlikely), but we shouldn't crash the IDE.
```

Comment:
📝 Visual Studio Isolated Shell no longer exists, so this code path is strictly non-null in test and production after this PR

Reply:
> Only if settings persistence service is not available in ISO shell and installing Roslyn as a component in ISO shell is supported. 

So we wouldn't have written this code and comment unless that actually happened at one point in the past.... :smile: